packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -147,10 +147,50 @@ export const askEnvironmentVariableCarryOut = async (context: $TSContext, curren
 
   await envVarQuestion();
   Object.keys(functions).forEach(functionName => {
-    setStoredKeyValue(context, functionName, functions);
+    setStoredKeyValue(context, functionName, functions[functionName]);
   });
 };
 
+export const askEnvironmentVariableToPush = async (context: $TSContext, currentEnvName: string, projectPath: string) => {
+  const teamProviderInfo = stateManager.getTeamProviderInfo(projectPath, {
+    throwIfNotExist: false,
+    default: {},
+  });
+  const functions = teamProviderInfo[currentEnvName]?.categories?.function || {};
+  const functionNames = Object.keys(functions);
+  if (functionNames.length === 0) {
+    return;
+  }
+  context.print.info('');
+  context.print.info('Some Lambda function environment variables are defined but missing values.');
+  context.print.info('');
+  for (const [_, functionName] of functionNames.entries()) {
+    const storedList = getStoredList(functionName);
+    const storedKeyValue = getStoredKeyValue(context, functionName);
+    for (const [_, item] of storedList.entries()) {
+      const envName = item.environmentVariableName;
+      const keyName = item.cloudFormationParameterName;
+      if (storedKeyValue.hasOwnProperty(keyName)) return;
+      const newValueQuestion: inquirer.InputQuestion = {
+        type: 'input',
+        name: 'newValue',
+        message: `Enter the missing environment variable value of ${envName} in ${functionName}:`,
+        validate: input => {
```

Comment:
would it be easy to reuse the same validator from the other env var value prompt here?